specfile/tags.py

```diff
@@ -0,0 +1,439 @@
+    "Posttransprog",
+    "Postun",
+    "Postunflags",
+    "Postunprog",
+    "Prefixes",
+    "Prein",
+    "Preinflags",
+    "Preinprog",
+    "Prereq",
+    "Pretrans",
+    "Pretransflags",
+    "Pretransprog",
+    "Preun",
+    "Preunflags",
+    "Preunprog",
+    "Provideflags",
+    "Providename",
+    "Providenevrs",
+    "Provides",
+    "Provideversion",
+    "Pubkeys",
+    "R",
+    "Recommendflags",
+    "Recommendname",
+    "Recommendnevrs",
+    "Recommends",
+    "Recommendversion",
+    "Recontexts",
+    "Release",
+    "Removepathpostfixes",
+    "Removetid",
+    "Requireflags",
+    "Requirename",
+    "Requirenevrs",
+    "Requires",
+    "Requireversion",
+    "Rpmversion",
+    "Rsaheader",
+    "Sha1header",
+    "Sha256header",
+    "Sig_base",
+    "Siggpg",
+    "Sigmd5",
+    "Sigpgp",
+    "Sigsize",
+    "Size",
+    "Source",
+    "Sourcepackage",
+    "Sourcepkgid",
+    "Sourcerpm",
+    "Suggestflags",
+    "Suggestname",
+    "Suggestnevrs",
+    "Suggests",
+    "Suggestversion",
+    "Summary",
+    "Supplementflags",
+    "Supplementname",
+    "Supplementnevrs",
+    "Supplements",
+    "Supplementversion",
+    "Transfiletriggerconds",
+    "Transfiletriggerflags",
+    "Transfiletriggerin",
+    "Transfiletriggerindex",
+    "Transfiletriggername",
+    "Transfiletriggerpostun",
+    "Transfiletriggerpriorities",
+    "Transfiletriggerscriptflags",
+    "Transfiletriggerscriptprog",
+    "Transfiletriggerscripts",
+    "Transfiletriggertype",
+    "Transfiletriggerun",
+    "Transfiletriggerversion",
+    "Triggerconds",
+    "Triggerflags",
+    "Triggerin",
+    "Triggerindex",
+    "Triggername",
+    "Triggerpostun",
+    "Triggerprein",
+    "Triggerscriptflags",
+    "Triggerscriptprog",
+    "Triggerscripts",
+    "Triggertype",
+    "Triggerun",
+    "Triggerversion",
+    "Url",
+    "V",
+    "Vcs",
+    "Vendor",
+    "Verbose",
+    "Verifyscript",
+    "Verifyscriptflags",
+    "Verifyscriptprog",
+    "Veritysignaturealgo",
+    "Veritysignatures",
+    "Version",
+    "Xpm",
+]
+
+
+class Tag:
+    """
+    Class that represents spec file tag.
+
+    Attributes:
+        name: Name of the tag.
+        value: Literal value of the tag as stored in the spec file.
+        comments: List of comment lines associated with the tag.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        value: str,
+        expanded_value: str,
+        separator: str,
+        preceding_lines: List[str],
+    ) -> None:
+        if not name or name.capitalize().rstrip("0123456789") not in TAG_NAMES:
+            raise ValueError(f"Invalid tag name: '{name}'")
+        self.name = name
+        self.value = value
+        self._expanded_value = expanded_value
+        self._separator = separator
+        self.comments, self._preceding_lines = self._extract_comments(preceding_lines)
+
+    def __repr__(self) -> str:
+        preceding_lines = repr(self._preceding_lines)
```

Comment:
Hm, this doesn't include comments does it? As far as I understand, the input of constructor `preceding_lines` includes all lines before the tag, comments and possibly something "extra" (maybe macro definition?). As far as I can see, this makes this `repr` behave incorrectly, for example if `preceding_lines=['%define macro macro', '# comment']` then the `_preceding_lines` attribute becomes only `['%define macro macro']` and hence the `repr` magic method doesn't actually return the correct way of constructing the object. Am I missing something?

Reply:
Good catch, it used to include comments as well, but I changed that and forgot to account for it in `__repr__()`.